core/src/main/scala/sttp/tapir/Validator.scala

```diff
@@ -139,6 +140,14 @@ object Validator extends ValidatorMagnoliaDerivation with ValidatorEnumMacro {
     }
   }
 
+  case class CustomCaseClass[T](doValidate: T => List[InvalidField[_]]) extends Single[T] {
```

Comment:
maybe we can simply change the `Custom` validator so that it accepts a `doValidate: T= > List[InvalidValue]` function?

Reply:
Wouldn't that require executing `doValidate` function twice - one for validation and second time for calculating error message

Reply:
yeah, we need to design it so that we avoid calling it twice :)